src/Microsoft.SqlTools.ServiceLayer/EditData/EditDataService.cs

```diff
@@ -0,0 +1,328 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Concurrent;
+using System.Data.Common;
+using System.Threading.Tasks;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.EditData.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Hosting;
+using Microsoft.SqlTools.ServiceLayer.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution.Contracts.ExecuteRequests;
+using Microsoft.SqlTools.ServiceLayer.Utility;
+using ConnectionType = Microsoft.SqlTools.ServiceLayer.Connection.ConnectionType;
+
+namespace Microsoft.SqlTools.ServiceLayer.EditData
+{
+    public class EditDataService : IDisposable
+    {
+        #region Singleton Instance Implementation
+
+        private static readonly Lazy<EditDataService> LazyInstance = new Lazy<EditDataService>(() => new EditDataService());
+
+        public static EditDataService Instance => LazyInstance.Value;
+
+        private EditDataService()
+        {
+            queryExecutionService = QueryExecutionService.Instance;
+            connectionService = ConnectionService.Instance;
+            metadataFactory = new SmoEditMetadataFactory();
+        }
+
+        internal EditDataService(QueryExecutionService qes, ConnectionService cs, IEditMetadataFactory factory)
+        {
+            queryExecutionService = qes;
+            connectionService = cs;
+            metadataFactory = factory;
+        }
+
+        #endregion
+
+        #region Member Variables 
+
+        private readonly ConnectionService connectionService;
+
+        private readonly IEditMetadataFactory metadataFactory;
+
+        private readonly QueryExecutionService queryExecutionService;
+
+        private readonly Lazy<ConcurrentDictionary<string, Session>> editSessions = new Lazy<ConcurrentDictionary<string, Session>>(
+            () => new ConcurrentDictionary<string, Session>());
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Dictionary mapping OwnerURIs to active sessions
+        /// </summary>
+        internal ConcurrentDictionary<string, Session> ActiveSessions => editSessions.Value;
+
+        #endregion
+
+        /// <summary>
+        /// Initializes the edit data service with the service host
+        /// </summary>
+        /// <param name="serviceHost">The service host to register commands/events with</param>
+        public void InitializeService(ServiceHost serviceHost)
```

Comment:
You don't initialize this in Program.cs. You will need to update it there. Or as I mentioned, make this an IHostedService like the FormatterService and it'll "just work".